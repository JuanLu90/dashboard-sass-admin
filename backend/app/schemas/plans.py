from pydantic import BaseModel
from typing import Dict, List

class Plan(BaseModel):
    name: str
    price: int
    period: str
    users: int
    description: str
    features: Dict[str, bool]

class PlansResponse(BaseModel):
    plans: List[Plan]
    featureList: List[str]
