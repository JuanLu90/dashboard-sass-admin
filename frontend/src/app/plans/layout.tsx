import { cookies } from "next/headers";
import { redirect } from "next/navigation";

export default function PlansLayout({ children }: { children: React.ReactNode }) {
  const token = cookies().get("token")?.value;

  if (!token) {
    redirect("/login");
  }

  return <>{children}</>;
}
