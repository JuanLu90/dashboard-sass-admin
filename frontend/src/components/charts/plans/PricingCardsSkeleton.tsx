import { Skeleton } from "@/components/ui/skeleton";

export default function PricingCardsSkeleton() {
  return (
    <div className="grid grid-cols-1 md:grid-cols-3 gap-6 mb-10">
      {[...Array(3)].map((_, i) => (
        <div
          key={i}
          className="bg-gray-900 rounded-xl shadow-md border border-gray-700 flex flex-col items-center p-8"
        >
          <Skeleton className="h-12 w-12 mb-4 rounded-full" />
          <Skeleton className="h-6 w-24 mb-2" />
          <Skeleton className="h-4 w-62 mb-2" />
          <Skeleton className="h-4 w-40 mb-6" />
          <Skeleton className="h-8 w-20 mb-6" />
          <Skeleton className="h-4 w-32 mb-6" />
          <Skeleton className="h-10 w-full rounded-md" />
        </div>
      ))}
    </div>
  );
}
