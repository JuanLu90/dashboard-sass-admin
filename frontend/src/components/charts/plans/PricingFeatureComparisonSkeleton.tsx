import { Skeleton } from "@/components/ui/skeleton";

export default function PricingFeatureComparisonSkeleton() {
  return (
    <div className="overflow-x-auto bg-gray-900 rounded-xl border border-gray-700 p-4">
      <table className="w-full text-left">
        <thead>
          <tr>
            <th className="px-4 py-3 font-semibold text-gray-300">Features</th>
            {[...Array(3)].map((_, i) => (
              <th key={i} className="px-4 py-3">
                <Skeleton className="h-6 w-24" />
              </th>
            ))}
          </tr>
        </thead>
        <tbody>
          {[...Array(4)].map((_, rowIndex) => (
            <tr key={rowIndex} className="border-t border-gray-700">
              <td className="px-4 py-2">
                <Skeleton className="h-5 w-32" />
              </td>
              {[...Array(3)].map((_, colIndex) => (
                <td key={colIndex} className="px-4 py-2 text-center">
                  <Skeleton className="h-5 w-5 mx-auto rounded-full" />
                </td>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
}
