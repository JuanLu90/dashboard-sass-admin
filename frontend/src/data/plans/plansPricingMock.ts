export const plansPricing = [
  {
    name: "Starter",
    price: 99,
    period: "yr",
    users: 1,
    description: "If you are a small business, please select this plan",
    features: {
      "Alias record": true,
      "Anycast DNS": true,
      "Full API access": false,
      "Multiple-layered DoS defense": false,
    },
  },
  {
    name: "Pro",
    price: 299,
    period: "yr",
    users: 5,
    description: "If you are a small business, please select this plan",
    features: {
      "Alias record": true,
      "Anycast DNS": true,
      "Full API access": true,
      "Multiple-layered DoS defense": false,
    },
  },
  {
    name: "Enterprise",
    price: 599,
    period: "yr",
    users: 20,
    description: "If you are a small business, please select this plan",
    features: {
      "Alias record": true,
      "Anycast DNS": true,
      "Full API access": true,
      "Multiple-layered DoS defense": true,
    },
  },
] as const;

export const featureList = [
  "Alias record",
  "Anycast DNS",
  "Full API access",
  "Multiple-layered DoS defense",
] as const;
