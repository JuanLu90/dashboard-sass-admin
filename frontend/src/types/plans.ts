export type FeatureName =
  | "Alias record"
  | "Anycast DNS"
  | "Full API access"
  | "Multiple-layered DoS defense";

export interface Plan {
  name: string;
  price: number;
  period: string;
  users: number;
  description: string;
  features: Record<FeatureName, boolean>;
}

export interface PlansResponse {
  plans: Plan[];
  featureList: FeatureName[];
}
