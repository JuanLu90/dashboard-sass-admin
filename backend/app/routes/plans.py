from fastapi import APIRouter
from app.schemas.plans import PlansResponse
from app.services.plans_service import get_plans_data
from app.utils.delay import simulate_delay

router = APIRouter()

@router.get("/plans", response_model=PlansResponse)
async def get_plans():
    await simulate_delay()
    return get_plans_data()
