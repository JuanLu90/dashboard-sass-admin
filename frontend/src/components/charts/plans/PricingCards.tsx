"use client";

import { Button } from "@/components/ui/button";
import { Medal, Award, Trophy } from "lucide-react";
import { apiFetch } from "@/lib/apiClient";
import { PlansResponse } from "@/types/plans";
import React from "react";

const icons = {
  Starter: <Award size={48} className="text-indigo-500 mb-4" />,
  Pro: <Medal size={48} className="text-indigo-500 mb-4" />,
  Enterprise: <Trophy size={48} className="text-indigo-500 mb-4" />,
};

export default function PricingCards() {
  const [data, setData] = React.useState<PlansResponse | null>(null);
  const [loading, setLoading] = React.useState(true);

  React.useEffect(() => {
    apiFetch<PlansResponse>("/api/plans")
      .then((plans) => setData(plans))
      .catch((err) => console.error("Failed to fetch plans", err))
      .finally(() => setLoading(false));
  }, []);

  if (!data || loading) return <p>Loading...</p>;

  return (
    <div className="grid grid-cols-1 md:grid-cols-3 gap-6 mb-10">
      {data.plans.map((plan) => (
        <div
          key={plan.name}
          className="bg-gray-900 rounded-xl shadow-md border border-gray-700 flex flex-col items-center p-8"
        >
          {icons[plan.name as keyof typeof icons]}
          <h3 className="text-2xl font-bold text-white mb-1">{plan.name}</h3>
          <div className="text-gray-400 mb-4 text-center">{plan.description}</div>
          <div className="text-3xl font-bold mb-2 text-white">
            ${plan.price}
            <span className="text-base font-normal text-gray-400">/{plan.period}</span>
          </div>
          <div className="text-gray-400 mb-6">
            {plan.users} User{plan.users > 1 ? "s" : ""}, Billed Yearly
          </div>
          <Button
            className="w-full mt-auto bg-indigo-600 text-white hover:bg-indigo-700"
            variant="default"
          >
            Select Plan
          </Button>
        </div>
      ))}
    </div>
  );
}
