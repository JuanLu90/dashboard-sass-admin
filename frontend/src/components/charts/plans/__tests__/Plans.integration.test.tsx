import { render, screen } from "@testing-library/react";
import PricingFeatureComparison from "../PricingFeatureComparison";
import { plansPricing, featureList } from "@/tests/mocks/plans/plansPricingMock";
import userEvent from "@testing-library/user-event";
import PricingCards from "../PricingCards";

describe("PricingFeatureComparison", () => {
  it("render plan`s names and features", () => {
    render(<PricingFeatureComparison />);

    plansPricing.forEach((plan) => {
      expect(screen.getByText(plan.name)).toBeInTheDocument();
    });

    featureList.forEach((feature) => {
      expect(screen.getByText(feature)).toBeInTheDocument();
    });
  });

  it("show check or X correctly by each feature and plan", () => {
    render(<PricingFeatureComparison />);

    const rows = screen.getAllByRole("row");
    const apiAccessRow = rows.find((row) => row.textContent?.includes("Full API access"));
    expect(apiAccessRow).toBeTruthy();
    expect(apiAccessRow?.children[1].querySelector(".text-gray-500")).toBeInTheDocument();
    expect(apiAccessRow?.children[2].querySelector(".text-green-400")).toBeInTheDocument();
    expect(apiAccessRow?.children[3].querySelector(".text-green-400")).toBeInTheDocument();
  });

  it("match snapshot", () => {
    const { container } = render(<PricingFeatureComparison />);
    expect(container).toMatchSnapshot();
  });
});

describe("PricingCards", () => {
  it("match snapshot", () => {
    const { container } = render(<PricingCards />);
    expect(container).toMatchSnapshot();
  });

  it("allow to click in Select Plan", () => {
    render(<PricingCards />);
    const buttons = screen.getAllByRole("button", { name: /select plan/i });
    userEvent.click(buttons[0]);
  });
});
