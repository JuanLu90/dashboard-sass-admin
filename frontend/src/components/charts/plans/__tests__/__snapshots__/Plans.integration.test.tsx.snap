// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PricingCards match snapshot 1`] = `
<div>
  <div
    class="grid grid-cols-1 md:grid-cols-3 gap-6 mb-10"
  >
    <div
      class="bg-gray-900 rounded-xl shadow-md border border-gray-700 flex flex-col items-center p-8"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-award text-indigo-500 mb-4"
        fill="none"
        height="48"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m15.477 12.89 1.515 8.526a.5.5 0 0 1-.81.47l-3.58-2.687a1 1 0 0 0-1.197 0l-3.586 2.686a.5.5 0 0 1-.81-.469l1.514-8.526"
        />
        <circle
          cx="12"
          cy="8"
          r="6"
        />
      </svg>
      <h3
        class="text-2xl font-bold text-white mb-1"
      >
        Starter
      </h3>
      <div
        class="text-gray-400 mb-4 text-center"
      >
        If you are a small business, please select this plan
      </div>
      <div
        class="text-3xl font-bold mb-2 text-white"
      >
        $
        99
        <span
          class="text-base font-normal text-gray-400"
        >
          /
          yr
        </span>
      </div>
      <div
        class="text-gray-400 mb-6"
      >
        1
         User
        , Billed Yearly
      </div>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive shadow-xs h-9 px-4 py-2 has-[>svg]:px-3 w-full mt-auto bg-indigo-600 text-white hover:bg-indigo-700"
        data-slot="button"
      >
        Select Plan
      </button>
    </div>
    <div
      class="bg-gray-900 rounded-xl shadow-md border border-gray-700 flex flex-col items-center p-8"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-medal text-indigo-500 mb-4"
        fill="none"
        height="48"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.21 15 2.66 7.14a2 2 0 0 1 .13-2.2L4.4 2.8A2 2 0 0 1 6 2h12a2 2 0 0 1 1.6.8l1.6 2.14a2 2 0 0 1 .14 2.2L16.79 15"
        />
        <path
          d="M11 12 5.12 2.2"
        />
        <path
          d="m13 12 5.88-9.8"
        />
        <path
          d="M8 7h8"
        />
        <circle
          cx="12"
          cy="17"
          r="5"
        />
        <path
          d="M12 18v-2h-.5"
        />
      </svg>
      <h3
        class="text-2xl font-bold text-white mb-1"
      >
        Pro
      </h3>
      <div
        class="text-gray-400 mb-4 text-center"
      >
        If you are a small business, please select this plan
      </div>
      <div
        class="text-3xl font-bold mb-2 text-white"
      >
        $
        299
        <span
          class="text-base font-normal text-gray-400"
        >
          /
          yr
        </span>
      </div>
      <div
        class="text-gray-400 mb-6"
      >
        5
         User
        s
        , Billed Yearly
      </div>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive shadow-xs h-9 px-4 py-2 has-[>svg]:px-3 w-full mt-auto bg-indigo-600 text-white hover:bg-indigo-700"
        data-slot="button"
      >
        Select Plan
      </button>
    </div>
    <div
      class="bg-gray-900 rounded-xl shadow-md border border-gray-700 flex flex-col items-center p-8"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-trophy text-indigo-500 mb-4"
        fill="none"
        height="48"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 14.66v1.626a2 2 0 0 1-.976 1.696A5 5 0 0 0 7 21.978"
        />
        <path
          d="M14 14.66v1.626a2 2 0 0 0 .976 1.696A5 5 0 0 1 17 21.978"
        />
        <path
          d="M18 9h1.5a1 1 0 0 0 0-5H18"
        />
        <path
          d="M4 22h16"
        />
        <path
          d="M6 9a6 6 0 0 0 12 0V3a1 1 0 0 0-1-1H7a1 1 0 0 0-1 1z"
        />
        <path
          d="M6 9H4.5a1 1 0 0 1 0-5H6"
        />
      </svg>
      <h3
        class="text-2xl font-bold text-white mb-1"
      >
        Enterprise
      </h3>
      <div
        class="text-gray-400 mb-4 text-center"
      >
        If you are a small business, please select this plan
      </div>
      <div
        class="text-3xl font-bold mb-2 text-white"
      >
        $
        599
        <span
          class="text-base font-normal text-gray-400"
        >
          /
          yr
        </span>
      </div>
      <div
        class="text-gray-400 mb-6"
      >
        20
         User
        s
        , Billed Yearly
      </div>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive shadow-xs h-9 px-4 py-2 has-[>svg]:px-3 w-full mt-auto bg-indigo-600 text-white hover:bg-indigo-700"
        data-slot="button"
      >
        Select Plan
      </button>
    </div>
  </div>
</div>
`;

exports[`PricingFeatureComparison match snapshot 1`] = `
<div>
  <div
    class="overflow-x-auto bg-gray-900 rounded-xl border border-gray-700"
  >
    <table
      class="w-full text-left"
    >
      <thead>
        <tr>
          <th
            class="px-4 py-3 font-semibold text-gray-300"
          >
            Features
          </th>
          <th
            class="px-4 py-3 font-semibold text-gray-300"
          >
            Starter
          </th>
          <th
            class="px-4 py-3 font-semibold text-gray-300"
          >
            Pro
          </th>
          <th
            class="px-4 py-3 font-semibold text-gray-300"
          >
            Enterprise
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="border-t border-gray-700 last:border-b"
        >
          <td
            class="px-4 py-2 font-medium text-gray-400"
          >
            Alias record
          </td>
          <td
            class="px-4 py-2 text-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-check text-green-400 mx-auto"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
          </td>
          <td
            class="px-4 py-2 text-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-check text-green-400 mx-auto"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
          </td>
          <td
            class="px-4 py-2 text-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-check text-green-400 mx-auto"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="border-t border-gray-700 last:border-b"
        >
          <td
            class="px-4 py-2 font-medium text-gray-400"
          >
            Anycast DNS
          </td>
          <td
            class="px-4 py-2 text-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-check text-green-400 mx-auto"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
          </td>
          <td
            class="px-4 py-2 text-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-check text-green-400 mx-auto"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
          </td>
          <td
            class="px-4 py-2 text-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-check text-green-400 mx-auto"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="border-t border-gray-700 last:border-b"
        >
          <td
            class="px-4 py-2 font-medium text-gray-400"
          >
            Full API access
          </td>
          <td
            class="px-4 py-2 text-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-x text-gray-500 mx-auto"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </td>
          <td
            class="px-4 py-2 text-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-check text-green-400 mx-auto"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
          </td>
          <td
            class="px-4 py-2 text-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-check text-green-400 mx-auto"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="border-t border-gray-700 last:border-b"
        >
          <td
            class="px-4 py-2 font-medium text-gray-400"
          >
            Multiple-layered DoS defense
          </td>
          <td
            class="px-4 py-2 text-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-x text-gray-500 mx-auto"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </td>
          <td
            class="px-4 py-2 text-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-x text-gray-500 mx-auto"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </td>
          <td
            class="px-4 py-2 text-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-check text-green-400 mx-auto"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
